### PR TITLE
Advi use updates

### PR DESCRIPTION
**Pull request by twiecki**
Continuation of #1822. CC @bhargavvader.

### CODE REVIEW

**Review by springcoil — COMMENTED** [review]
All looks good to me, no major stylistic recommendations. I think others can comment in more depth though.

File: pymc3/variational/advi.py [hunk 1]
```diff
@@ -47,7 +48,7 @@ def advi(vars=None, start=None, model=None, n=5000, accurate_elbo=False,
     The variational parameters are defined on the transformed space, which is
     required to do ADVI on an unconstrained parameter space as described in
     [KTR+2016]. The parameters in the :code:`ADVIfit` object are in the
-    transformed space, while traces returned by :code:`sample_vp()` are in
+    transformed space, while traces preturned by :code:`sample_vp()` are in
```
Comment (springcoil) [inline pymc3/variational/advi.py]: `preturned` is this a word?

File: pymc3/variational/advi.py [hunk 2]
```diff
@@ -138,7 +138,8 @@ def advi(vars=None, start=None, model=None, n=5000, accurate_elbo=False,
     # Create parameter update function used in the training loop
     uw_shared = theano.shared(uw, 'uw_shared')
     elbo = pm.CallableTensor(elbo)(uw_shared)
-    updates = optimizer(-1 * elbo, [uw_shared], learning_rate=learning_rate, epsilon=epsilon)
+    updates = optimizer([-1 * elbo], [uw_shared], learning_rate=learning_rate, epsilon=epsilon)
+    updates = apply_momentum(updates, [uw_shared], momentum=0.9)
```
Comment (taku-y) [inline pymc3/variational/advi.py]: Always use momentum? I think we can disable mometum if you want stability in optimization.

Comment (twiecki) [reply pymc3/variational/advi.py]: You're right, I assumed the previous implementation always used momentum but that's not the case.

**Comment by bhargavvader** [comment]
What are the results looking like? Does it match the old optimiser results or are the tests failing like the old PR?

**Comment by twiecki** [comment]
@bhargavvader Still failing, alas. I found one critical error though, one needs to pass in a list of gradients, as updates.py otherwise thinks a tensor represents the loss function and will calculate its gradient again.

**Comment by twiecki** [comment]
Maybe removing the momentum helps.
